x-pack/plugins/actions/server/usage/actions_usage_collector.ts

```diff
@@ -16,6 +16,13 @@ export function createActionsUsageCollector(
   return usageCollection.makeUsageCollector<ActionsUsage>({
     type: 'actions',
     isReady: () => true,
+    schema: {
+      count_total: { type: 'long' },
+      count_active_total: { type: 'long' },
+      // TODO: Find out all the possible values for DYNAMIC_KEY or reformat into an array
+      count_by_type: { DYNAMIC_KEY: { type: 'long' } },
+      count_active_by_type: { DYNAMIC_KEY: { type: 'long' } },
```

Comment:
It looks like we can get a list of types here since the collector is created after `setup` phase (https://github.com/elastic/kibana/blob/master/x-pack/plugins/actions/server/plugin.ts#L228). Did you need some help to access the list of types?

Reply:
> Also not sure how this would then correlate to `x-pack/plugins/telemetry_collection_xpack/schema/xpack_plugins.json`.

We run the command `node scripts/telemetry_extract.js` to find all the `schema` declarations and merge them into that file.

> It looks like we can get a list of types here since the collector is created after `setup` phase ([`master`/x-pack/plugins/actions/server/plugin.ts#L228](https://github.com/elastic/kibana/blob/master/x-pack/plugins/actions/server/plugin.ts?rgh-link-date=2020-09-30T12%3A51%3A50Z#L228)). Did you need some help to access the list of types?

Unfortunately, the command mentioned above doesn't run Kibana. It scans the code instead. But if you can point me to way other plugins register actions, I can search them and populate the current list. We'll likely need to refresh the list on every FF, until we find a way to automate it. Is it just looking for `registerType(ACTION_TYPE)`?

Reply:
> Is it just looking for `registerType(ACTION_TYPE)`?

I've looked for those and I couldn't find any use of those (only in test plugins) 🤔

Reply:
Ah, it would be because only the actions plugin registers action types at this time. They are all defined [here](https://github.com/elastic/kibana/blob/master/x-pack/plugins/actions/server/builtin_action_types/index.ts#L30-L38) but call `register(...)` which is a lower level function.

https://github.com/elastic/kibana/blob/master/x-pack/plugins/actions/server/builtin_action_types/index.ts#L30-L38

Interesting about scanning the code. You will find plugins registering alert types but we don't have any yet registering action types (in 1-2 release from now maybe).

Reply:
@mikecote thank you for the pointer! I'll update the list with the known keys.

Similar to the question in `alerts`, do you think it's still a good idea to maintain the `DYNAMIC_KEY` to let the Remote Telemetry cluster know that it might grow (just in case they want to set up a dynamic mapping template of some sort)?

Reply:
@mikecote done! I just pushed the updated schema with the list of known built-in actions. Thank you for pointing me to where to find them